packages/cli/src/commands/source/push/add.ts

```diff
@@ -0,0 +1,154 @@
+import {Source} from '@coveo/push-api-client';
+import {Command, flags} from '@oclif/command';
+import {blueBright, green} from 'chalk';
+import {readdirSync} from 'fs';
+import path from 'path';
+import dedent from 'ts-dedent';
+import {
+  buildAnalyticsFailureHook,
+  buildAnalyticsSuccessHook,
+} from '../../../hooks/analytics/analytics';
+import {
+  IsAuthenticated,
+  Preconditions,
+} from '../../../lib/decorators/preconditions';
+import {AuthenticatedClient} from '../../../lib/platform/authenticatedClient';
+import {parseAndGetDocumentBuilderFromJSONDocument} from '../../../lib/push/parseFile';
+
+interface AxiosResponse {
+  status: number;
+  statusText: string;
+}
+
+export default class SourcePushAdd extends Command {
+  public static description = 'Push a JSON document into a Coveo push source.';
+
+  public static flags = {
+    file: flags.string({
+      multiple: true,
+      exclusive: ['folder'],
+      char: 'f',
+      helpValue: 'myfile.json',
+      description: 'One or multiple file to push. Can be repeated.',
+    }),
+    folder: flags.string({
+      multiple: true,
+      exclusive: ['file'],
+      char: 'd',
+      helpValue: './my_folder_with_multiple_json_files',
+      description:
+        'One or multiple folder containing json files. Can be repeated',
+    }),
+  };
+
+  public static args = [
+    {
+      name: 'sourceId',
+      required: true,
+      description:
+        'The identifier of the source on which to perform the add operation. See source:push:list to obtain the identifier.',
+    },
+  ];
+
+  @Preconditions(IsAuthenticated())
+  public async run() {
+    const {flags} = this.parse(SourcePushAdd);
+    if (!flags.file && !flags.folder) {
+      this.error(
+        'You must provide either --file= or --folder=. See source:push:add --help for more information.'
+      );
+    }
+    const cfg = await new AuthenticatedClient().cfg.get();
+    const source = new Source(cfg.accessToken!, cfg.organization);
+
+    if (flags.file) {
+      await this.doPushFile(source);
+    }
+    if (flags.folder) {
+      await this.doPushFolder(source);
+    }
```

Comment:
It should be either file or folder, right?

If so I would do something like that:
```suggestion
    const cfg = await new AuthenticatedClient().cfg.get();
    const source = new Source(cfg.accessToken!, cfg.organization);
    let isFolderXORFileSet = false;
    if (flags.file) {
      await this.doPushFile(source);
      isFolderOrFileSet = !isFolderOrFileSet;
    }
    if (flags.folder) {
      await this.doPushFolder(source);
      isFolderOrFileSet = !isFolderOrFileSet;
    }
    if (!isFolderXORFileSet) {
      this.error(
        'You must provide either --file= or --folder=. See source:push:add --help for more information.'
      );
    }
```

Reply:
It's the same end result though 🤔 Why do you prefer this approach ? Just curious.

Reply:
I find the initial way easier to read.

Reply:
I will leave it as is then. 

If there's any strong objection, please ping back.